wallets/core/src/types.ts

```diff
@@ -74,7 +74,7 @@ export type Subscribe = (options: {
   updateAccounts: (accounts: string[], chainId?: string) => void;
   connect: (network?: Network) => void;
   disconnect: () => void;
-}) => void;
+}) => () => void;
```

Comment:
if a `subscribe` function doesn't implement a cleanup, it will return `undefined` by default in JS. So I think the type here will be something like this:
`
type CleanupSubscribe = () => void;

// ... 
=> CleanupSubscribe | undefined
`

Reply:
fixed.